sky/backends/backend_utils.py

```diff
@@ -1106,9 +1124,18 @@ def wait_until_ray_cluster_ready(
             cluster_config_file, max_attempts=WAIT_HEAD_NODE_IP_MAX_ATTEMPTS)
     except RuntimeError as e:
         logger.error(e)
-        return False  # failed
+        return False, None  # failed
+
+    docker_user = None
+    if use_docker:
+        # pylint: disable=import-outside-toplevel
+        from sky.backends.docker_utils import docker_host_setup
+        docker_user = docker_host_setup(head_ip, cluster_config_file)
```

Comment:
Quick answer: this option requires the docker image containing a user named `my_username`, which is typically not the case for our default username `gcpuser` or `azureuser`. But this is the implementation previous to overriding Ray's `DockerCommandRunner`; since now we have the ability to modify the docker launch process, there may be some workarounds available and I'm trying it.